packages/zoe/src/contracts/otcDeskReputation.js

```diff
@@ -0,0 +1,142 @@
+          keyword,
+          amount.brand,
+        );
+        return zcf.getAmountMath(amount.brand).isGTE(allocatedAmount, amount);
+      });
+      if (!hasInventory) {
+        numTradesDefected += 1;
+        throw Error(
+          `The market maker did not have the inventory they promised. Their rating has been adjusted accordingly`,
+        );
+      }
+
+      // If this trade does not conserve rights, it will throw an
+      // error and the user will get their funds back.
+      try {
+        trade(
+          zcf,
+          {
+            seat: marketMakerSeat,
+            gains: price,
+            losses: assets,
+          },
+          { seat, gains: userWantActual, losses: userGiveActual },
+        );
+      } catch (err) {
+        console.log('actual proposal: ', seat.getProposal());
+        console.log('quote: ', {
+          assets,
+          price,
+        });
+        throw Error(
+          'Your offer did not match the quote. See the console log for more details.',
+        );
+      }
+      seat.exit();
+      return 'Trade successful';
+    };
+
+    const customProperties = harden({
+      underlyingAssets: assets,
+      strikePrice: price,
+      byTimerAuthority: timerAuthority,
+      beforeOrAtDeadline: deadline,
+    });
+
+    const acceptQuoteInvitation = zcf.makeInvitation(
+      acceptQuote,
+      'acceptQuote',
+      customProperties,
+    );
+    return acceptQuoteInvitation;
+  };
+
+  const addInventory = seat => {
+    // Take everything in this seat and add it to the marketMakerSeat
+    trade(
+      zcf,
+      { seat: marketMakerSeat, gains: seat.getCurrentAllocation() },
+      { seat, gains: {} },
+    );
+    seat.exit();
+    return 'Inventory added';
+  };
+
+  const removeInventory = seat => {
+    const { want } = seat.getProposal();
+    trade(zcf, { seat: marketMakerSeat, gains: {} }, { seat, gains: want });
+    seat.exit();
+    return 'Inventory removed';
+  };
+
+  const creatorFacet = {
+    // The inventory can be added in bulk before any quotes are made
+    // or can be added immediately before a quote.
+    makeAddInventoryInvitation: async (issuerKeywordRecord = undefined) => {
+      const { issuers } = zcf.getTerms();
+      const issuersPSaved = Object.entries(issuerKeywordRecord).map(
+        ([keyword, issuer]) => {
+          // If the keyword does not yet exist, add it and the
+          // associated issuer.
+          if (issuers.keyword === undefined) {
+            return zcf.saveIssuer(issuer, keyword);
+          }
+          return undefined;
+        },
+      );
+      await Promise.all(issuersPSaved);
+      return zcf.makeInvitation(addInventory, 'addInventory');
+    },
+    makeRemoveInventoryInvitation: () =>
+      zcf.makeInvitation(removeInventory, 'removeInventory'),
+    makeQuote,
+  };
+
+  const publicFacet = {
+    getRating: () =>
+      numTradesAttempted > 0
+        ? `${Math.floor(
```

Comment:
Might return the absolute numbers as well as the rating `{ rating, numAttempts, numDefects }`.  Or the timestamp of last failure. "It has been [17 days] since I burned a customer"